packages/opentelemetry-exporter-stackdriver-trace/src/trace.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExportResult } from '@opentelemetry/base';
+import { NoopLogger } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { Logger } from '@opentelemetry/types';
+import { GoogleAuth } from 'google-auth-library';
+import { google } from 'googleapis';
+import { StackdriverExporterOptions } from './external-types';
+import { getReadableSpanTransformer } from './transform';
+import { Span, SpansWithCredentials } from './types';
+
+google.options({ headers: { 'x-opentelemetry-outgoing-request': 1 } });
+
+/**
+ * Format and sends span information to StackDriver Trace.
+ */
+export class StackdriverTraceExporter implements SpanExporter {
+  private _projectId?: string;
+  private _serviceName: string;
+  private _logger: Logger;
+  private _auth: GoogleAuth;
+  private static _cloudTrace = google.cloudtrace('v2');
+
+  constructor(options: StackdriverExporterOptions) {
+    this._logger = options.logger || new NoopLogger();
+    this._projectId = options.projectId;
```

Comment:
How to handle missing project id? right now it just fails to export

Reply:
It should ideally pull the project ID from the instance metadata when running on GCP (GCE, GKE, App Engine, Cloud Run, etc.). When not running on GCP, I think we can still get the project ID from the credential metadata (private key .json file)?

Reply:
I can apply these, but there is still the possibility that it fails. If we cannot get the project id, should we throw on start or just fail to export? Right now, the export fails and logs (without throwing).